dart/lib/src/hub.dart

```diff
@@ -81,7 +81,19 @@ class Hub {
       );
     } else {
       final item = _peek();
-      final scope = await _cloneAndRunWithScope(item.scope, withScope);
+      final Scope scope;
+
+      try {
+        scope = await _cloneAndRunWithScope(item.scope, withScope);
+      } catch (exception, stackTrace) {
+        _options.logger(
+          SentryLevel.error,
+          'Failed to capture event with id: ${event.eventId}',
+          exception: exception,
+          stackTrace: stackTrace,
+        );
+        return sentryId;
```

Comment:
I'd not drop the event, but rather log it and keep going.

Reply:
Sentry dotnet does return an empty id though, thats wehre i got the initial implementation from. But i agree that we can continue here.

https://github.com/getsentry/sentry-dotnet/blob/41acafb4b3bfced16b546dd7aa6900603b349ceb/src/Sentry/Internal/Hub.cs#L284